react/components/molecules/snackbar-ios/snackbar-ios.js

```diff
@@ -0,0 +1,125 @@
+import React, { PureComponent } from "react";
+import { Animated, Platform, StyleSheet, Text, ViewPropTypes } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { baseStyles, IdentifiableMixin } from "../../../util";
+
+import { Touchable } from "../../atoms";
+
+export class SnackbarIos extends mix(PureComponent).with(IdentifiableMixin) {
+    static get propTypes() {
+        return {
+            text: PropTypes.string,
+            actionText: PropTypes.string,
+            showDuration: PropTypes.number,
+            animationDuration: PropTypes.number,
+            onActionPress: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            text: undefined,
+            actionText: undefined,
+            showDuration: 5000,
+            animationDuration: 300,
+            onActionPress: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            opacity: new Animated.Value(0),
+            snackbarTimeout: null
+        };
+    }
+
+    _stopPrevAnimations = () => {
+        this.state.opacity.stopAnimation();
+        this.state.opacity.setValue(0);
+        clearTimeout(this.snackbarTimeout);
+    };
+
+    show = () => {
+        this._stopPrevAnimations();
+        Animated.timing(this.state.opacity, {
+            toValue: 1,
+            duration: this.props.animationDuration,
+            useNativeDriver: true
+        }).start();
+        this.snackbarTimeout = setTimeout(this.hide, this.props.showDuration);
+    };
+
+    hide = () => {
+        clearTimeout(this.snackbarTimeout);
+        Animated.timing(this.state.opacity, {
+            toValue: 0,
+            duration: this.props.animationDuration,
+            useNativeDriver: true
+        }).start();
+    };
+
+    _style = () => {
+        return [
+            styles.snackbar,
+            {
+                opacity: this.state.opacity
+            },
+            this.props.style
+        ];
+    };
+
+    render() {
+        return (
+            <Animated.View style={this._style()} {...this.id("snackbar-ios")}>
+                <Text style={styles.text}>{this.props.text}</Text>
+                {this.props.actionText && (
+                    <Touchable onPress={this.props.onActionPress} style={styles.action}>

```

Comment:
I believe that touching this should hide the snackbar before calling the callback passed by props.